web/api/v1/api_test.go

```diff
@@ -739,13 +739,14 @@ func TestLabelNames(t *testing.T) {
 	api := &API{
 		Queryable: storage,
 	}
-	request := func(method string, matchers ...string) (*http.Request, error) {
+	request := func(method, limit string, matchers ...string) (*http.Request, error) {
 		u, err := url.Parse("http://example.com")
 		require.NoError(t, err)
 		q := u.Query()
 		for _, matcher := range matchers {
 			q.Add("match[]", matcher)
 		}
+		q.Add("limit", limit)
```

Comment:
I'd add a `limit!=""` check before this so we don't change the previous behavior. (before we weren't setting the limit param, now we're setting it to the empty string, they should be handled the same way, but it's better no to change existing tests)